### PR TITLE
feat: add breadcrumbs to open-api-view

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds a breadcrumb bar to the top of the OpenAPI docs view.

## 📸 Design Screenshots

🎨 [Figma file][figma]

> **Note**: breadcrumbs in the implementation differ slightly from those in the design, as the URL structure is `HCP → API docs → Vault Secrets` rather than `HCP → Vault Secrets → API` as designed. We could potentially move the URL structure around to accommodate this, although this would require a redirect as https://developer.hashicorp.com/hcp/api-docs/vault-secrets is already live.

| Before | After |
| - | - |
| before | after |

## 🧪 Testing

- [ ] Visit [/hcp/api-docs/vault-secrets] in the preview
    - The breadcrumb bar should appear at the top of the page
    - Links in the breadcrumb bar should look & function as expected
    - The `API` breadcrumb item should be unlinked, as we have no `/hcp/api-docs` page to link to

## 💭 Anything else?

Not at the moment!

[figma]: https://www.figma.com/file/wjoC3ZVsA1kDuPsFLjEFuY/API-Template?type=design&node-id=526%3A33909&mode=design&t=CzoH0Mbr1tyw9w99-1
[task]: https://app.asana.com/0/1204678746647847/1205138989258548/f
[preview]: https://dev-portal-git-zsopen-api-view-breadcrumbs-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zsopen-api-view-breadcrumbs-hashicorp.vercel.app/hcp/api-docs/vault-secrets

### CODE REVIEW

File: src/views/open-api-docs-view/server.ts [hunk 1]
```diff
@@ -110,6 +108,12 @@ export async function getStaticProps({
 		productSlug: productData.slug,
 	})
 
+	/**
+	 * Build breadcrumb links for the page, and activate the final breadcrumb.
+	 */
+	const breadcrumbLinks = getBreadcrumbLinks(basePath)
```
Comment (zchsh) [inline src/views/open-api-docs-view/server.ts:117]: ✨ `getBreadcrumbLinks` in action!

**Comment by braydenlove** [comment]
Breadcrumbs look good, I think we worry about the structure via the IA work.

**Comment by braydenlove** [comment]
Can we confirm that the header area matches the mobile designs? Currently it is wrapping strangely. 
image
image
https://www.figma.com/file/wjoC3ZVsA1kDuPsFLjEFuY/API-Template?type=design&node-id=386%3A16963&mode=design&t=khhnZEFxBBcfCYwK-1

**Comment by zchsh** [comment]
@braydenlove Ah yeah, does look like the header area needs some mobile work! I'll handle that in a separate PR if that's alright - ticket here: 🎟️ [Refine mobile appearance of Overview section](https://app.asana.com/0/1204678746647847/1205278197175200/f)

**Comment by braydenlove** [comment]
Awesome @zchsh thanks!

**Review by braydenlove — APPROVED** [review]

**Comment by zchsh** [comment]
🤦 ack, forgot to revert flags before merging again... might just leave the preview flags on for now!

**Pull request merged**